### PR TITLE
Move node-perf-dash to new infrastructure.

### PR DESCRIPTION
**Pull request by lorqor**
https://github.com/kubernetes/k8s.io/issues/753

### CODE REVIEW

File: node-perf-dash.k8s.io/node-perf-dash-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
```
Comment (ameukam) [inline node-perf-dash.k8s.io/node-perf-dash-deployment.yaml:3]: namespace field need to be mentioned.

File: node-perf-dash.k8s.io/node-perf-dash-service.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+apiVersion: v1
+kind: Service
+metadata:
```
Comment (ameukam) [inline node-perf-dash.k8s.io/node-perf-dash-service.yaml:3]: namespace field need to be mentioned.

**Comment by bartsmykla** [comment]
/ok-to-test

@lorqor thank you for picking it up, I'll come back to you with few hints, as we are missing few things here

**Comment by bartsmykla** [comment]
@lorqor 
So what do we miss in this PR:

- Adding created rbac group (`k8s-infra-rbac-node-perf-dash@kubernetes.io`) to the `gke-security-groups@kubernetes.io` group
- Ingress file
- Certificate file with the staging subdomain (node-perf-dash-staging.k8s.io)

What should be done before this PR:

- Adding new static IP address
- Creating the temporary/staging DNS record which would point to the IP address created in the step above

What should be done after this PR:

- When there will be confirmation everything works at the staging subdomain, we can update the original record (node-perf-dash.k8s.io) and remove unnecessary anymore staging one (node-perf-dash-staging.k8s.io)

<hr>

More instructions below:

- Static IP address
  - Adding the name of the address to [`infra/gcp/ensure-static-ips.sh`](https://github.com/kubernetes/k8s.io/blob/master/infra/gcp/ensure-static-ips.sh#L39-L45) - I would suggest the name: `node-perf-dash-k8s-io-ingress-prod`
  - Somebody with permissions (member of [k8s-infra-cluster-admins@kubernetes.io](https://github.com/kubernetes/k8s.io/blob/master/groups/groups.yaml#L218-L234) group) need to run the script

- Staging DNS record
  - [Creating new DNS Update issue](https://github.com/kubernetes/k8s.io/issues/new?assignees=&labels=wg%2Fk8s-infra%2C+area%2Fdns&template=dns-request.md&title=DNS+REQUEST%3A+%3Cyour-dns-record%3E) which could look like:

    ```markdown
    ### Type of DNS update:
   Create

    ### Domain being modified:
    e.g. `k8s.io`

    ### New DNS Record:

    ```yaml
    node-perf-dash-staging.k8s.io:
    - type: A
      value: [IP_ADDRESS_CREATED_IN_STEP_BEFORE]
    ```

    ### Reason for update:

    Moving `node-perf-dash` to the `aaa` cluster
    ```

  - Creating the PR (or updating this one) with the new DNS record placed in config file: [`dns/zone-configs/k8s.io._0_base.yaml`](https://github.com/kubernetes/k8s.io/blob/master/dns/zone-configs/k8s.io._0_base.yaml)

  - When the PR will be approved, somebody with permissions (member of the google group: `k8s-infra-dns-admins@kubernetes.io`) have to reconcile the DNSes

- Ingress
  - Adding file: `node-perf-dash.k8s.io/ingress.yaml`

    ```yaml
    apiVersion: extensions/v1beta1
    kind: Ingress
    metadata:
    name: node-perf-dash
    labels:
        app: node-perf-dash
    annotations:
        kubernetes.io/ingress.global-static-ip-name: node-perf-dash-k8s-io-ingress-prod
    spec:
    tls:
    - secretName: node-perf-dash-k8s-io-tls
    backend:
        serviceName: node-perf-dash-status
        servicePort: status
    ```

  - Adding to `node-perf-dash.k8s.io/README.md` instruction

    ```shell
    kubectl apply -n node-perf-dash -f ingress.yaml
    ```

- Certificate
  - Adding file: `node-perf-dash.k8s.io/certificate.yaml`

    ```yaml
    apiVersion: cert-manager.io/v1alpha2
    kind: Certificate
    metadata:
      name: node-perf-dash-k8s-io
      annotations:
        acme.cert-manager.io/http01-override-ingress-name: "node-perf-dash"
        cert-manager.io/issue-temporary-certificate: "true"
    spec:
      secretName: node-perf-dash-k8s-io-tls
      issuerRef:
        kind: ClusterIssuer
        name: letsencrypt-prod
      dnsNames:
      - node-perf-dash-staging.k8s.io
      - node-perf-dash.k8s.io
    ```

  - Adding to `node-perf-dash.k8s.io/README.md` instruction

    ```shell
    kubectl apply -n node-perf-dash -f certificate.yaml
    ```

- Adding created rbac group (`k8s-infra-rbac-node-perf-dash@kubernetes.io`) to the `gke-security-groups@kubernetes.io` group

  - Modyfing the [`groups/groups.yaml`](https://github.com/kubernetes/k8s.io/blob/f9609997bbcbd91d2d9a103cfda823b5d039f7b3/groups/groups.yaml#L132-L147) file:

    ```yaml
    # Every RBAC group should be added here.
    - email-id: gke-security-groups@kubernetes.io
      name: gke-security-groups
      description: |-
        Security Groups for GKE clusters
      settings:
        ReconcileMembers: "true"
        WhoCanViewMembership: "ALL_MEMBERS_CAN_VIEW" # needed for RBAC
      members:
        - k8s-infra-rbac-cert-manager@kubernetes.io
        - k8s-infra-rbac-gcsweb@kubernetes.io
        - k8s-infra-rbac-k8s-io-canary@kubernetes.io
        - k8s-infra-rbac-k8s-io-prod@kubernetes.io
        - k8s-infra-rbac-perfdash@kubernetes.io
        - k8s-infra-rbac-publishing-bot@kubernetes.io
        - k8s-infra-rbac-slack-infra@kubernetes.io
        - k8s-infra-rbac-node-perf-dash@kubernetes.io
    ```

- When there will be confirmation everything works at the staging subdomain, we can update the original record (node-perf-dash.k8s.io) and remove unnecessary anymore staging one (node-perf-dash-staging.k8s.io)

  - Creating the DNS Update issue (see step 1) for the update of the `node-perf-dash.k8s.io` record
  - Creating the DNS Update issue (see step 1) for the removal ofthe  `node-perf-dash-staging.k8s.io` record
  - Creating PR with removal of the `node-perf-dash-staging.k8s.io` from the file: `node-perf-dash.k8s.io/certificate.yaml`

    ```yaml
    apiVersion: cert-manager.io/v1alpha2
    kind: Certificate
    metadata:
      name: node-perf-dash-k8s-io
      annotations:
        acme.cert-manager.io/http01-override-ingress-name: "node-perf-dash"
        cert-manager.io/issue-temporary-certificate: "true"
    spec:
      secretName: node-perf-dash-k8s-io-tls
      issuerRef:
        kind: ClusterIssuer
        name: letsencrypt-prod
      dnsNames:
      - node-perf-dash.k8s.io
    ```

**Comment by bartsmykla** [comment]
@lorqor:

as #867 was merged, and the IP was created I opened #869 and #870 to create the temporary/staging subdomain which should be used if everything works fine before we'll switch the proper subdomain

**Comment by lorqor** [comment]
@bartsmykla 

Hi Bart, thanks for reviewing the PR and providing detailed instructions for the migration and filing #869 and #870. This is very helpful for someone like me that has zero experience in contributing to the domain :) 

I added the ingress and certificate files in the patch. For `gke-security-groups@kubernetes.io`, I included it in the first commit in https://github.com/kubernetes/k8s.io/pull/864/files#diff-8b242825ad9027b6efead6a412418c51L133, is that the same thing you were talking about?

Thanks!

File: node-perf-dash.k8s.io/node-perf-dash-certificate.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    acme.cert-manager.io/http01-override-ingress-name: "node-perf-dash"
+    cert-manager.io/issue-temporary-certificate: "true"
+spec:
+  secretName: node-perf-dash-k8s-io-tls
+  issuerRef:
+    kind: ClusterIssuer
+    name: letsencrypt-prod
+  dnsNames:
+  - node-perf-dash-staging.k8s.io
+  - node-perf-dash.k8s.io
```
Comment (bartsmykla) [inline node-perf-dash.k8s.io/node-perf-dash-certificate.yaml:16]: nit: could you add the last line?

Comment (lorqor) [reply node-perf-dash.k8s.io/node-perf-dash-certificate.yaml]: Done. Thanks!

**Comment by bartsmykla** [comment]
/assign @dims 

@dims please, as an group OWNER look :-)

**Comment by lorqor** [comment]
We have a data race ¯\_(ツ)_/¯

**Comment by dims** [comment]
we have a hold pattern because of https://github.com/kubernetes/k8s.io/issues/852 let's please wait for it to get resolved.

**Comment by bartsmykla** [comment]
@lorqor I have deployed the `node-perf-dash` and it should be at [node-perf-dash-staging.k8s.io](https://node-perf-dash-staging.k8s.io) - please confirm if everything works as expected and that you have access to the `node-perf-dash` namespace at the `aaa` cluster (more info [here](https://github.com/kubernetes/k8s.io/blob/master/running-in-community-clusters.md#access-the-cluster)) and if so we can proceed and point the `node-perf-dash.k8s.io` to the new cluster and remove `node-perf-dash-staging.k8s.io`

**Pull request merged**